### DISCUSSION TITLE
What colours the line and column numbers in the diagnostic, lsp_references and lsp_definition pickers?

### DISCUSSION THREAD

**Discussion by artcodespace** [ibhagwan/fzf-lua#1153] [General]
I'm currently trying to produce a grey scale neovim setup that only uses colour for errors and warnings. Currently working on fzf-lua. My config is here: https://github.com/alunturner/.dotfiles/blob/b23791909559c50200feb6e107a43ba7d481cbeb/nvim/.config/nvim/lua/plugins/finder.lua#L23

I'm close, but I have the following issue. I can not remove the line (green) and column (blue) number colouring from the pickers mentioned in the title. The only way I've found of doing it is additionally using `fzf_opts` to set `--color=bw`. This works... but it also knocks out the colour in the diagnostics (which I'm trying to keep).

I've looked at the fzf-lua documentation and profiles, man fzf, man rg but I just can't pin down where it comes from. I suppose it could even be encoded in the lap output. 

Any suggestions for removing the blue and green much appreciated.
![PXL_20240428_195816297~2](https://github.com/ibhagwan/fzf-lua/assets/56027671/bb9302a5-c2dc-421a-89b2-9ba46289c92e)

**Comment by ibhagwan** [comment]
The line and column colors are hardcoded using blue and green terminal colors similar to the output of `rg` (which isn't controlled by fzf-lua):
https://github.com/ibhagwan/fzf-lua/blob/7a2d9849fe6ee87a386a8b9486ebd2e2884184fa/lua/fzf-lua/make_entry.lua#L254-L255

I suppose I could use `FzfLuaLineNr` for the line number and create a new highlight for `FzfLuaColNr`, shouldn't be a big deal to add.

**Comment by ibhagwan** [comment]
@alunturner,  https://github.com/ibhagwan/fzf-lua/commit/40cf21d8b66ce3793b389382f127e287222c4bc5 should solve your issue.

Default colors are linked to `LightGreen` and `CadetBlue1`:
![image](https://github.com/ibhagwan/fzf-lua/assets/59988195/f6735a7d-830f-4af1-9799-613da3b272eb)


You can either set these via regular hl links, e.g:
```lua
:hi! link FzfLuaPathLineNr ErrorMsg
:hi! link FzfLuaPathColNr IncSearch
```

Or via `setup.hls` or in the call directly:
```lua
:FzfLua quickfix hls.path_linenr=ErrorMsg hls.path_colnr=IncSearch
```
Will result in:
![image](https://github.com/ibhagwan/fzf-lua/assets/59988195/872234e5-6b5a-46ac-97ce-f2cfc8f607c8)

The new hl groups apply to all pickers where the hardcoded line/col were used (not in grep though as this is controlled by `rg`):
- buffers
- tabs
- lines|blines
- lsp
- diagnostics
- quickfix|loclist

**Comment by artcodespace** [comment]
Hi @ibhagwan, thanks so much for this!
I'll try it out when I get a chance today.

Thanks for the fix but thanks more generally for writing and maintaining this excellent plugin, I really enjoy using it.

**Reply by ibhagwan** [reply]
You’re welcome @alunturner and ty for your kind words!
